### PR TITLE
Android asset manager rehault - Part 1

### PR DESCRIPTION
**Pull request by selaux**
This fixes 2 out of 4 issues in #1248.

- Makes Android Asset VFS case insensitive
- Stores pointer to JNI globally in Rust

One thing to note is thatI needed to implement case insensitive matching via JNI, as it is not possible to list asset directories through the NDK asset manager. It can only list files in directories not directories.

Supporting mods from assets is gonna be a little trickier, as mods may also contain SLF files, so we have an SLF VFS within another VFS.

### CODE REVIEW

File: src/sgp/SGP.cc [hunk 1]
```diff
@@ -349,13 +349,12 @@ int main(int argc, char* argv[])
 
 		#ifdef __ANDROID__
 		JNIEnv* jniEnv = (JNIEnv*)SDL_AndroidGetJNIEnv();
-		if (jniEnv == NULL) {
-			SLOGE("Failed to get jni env");
+		
+		if (setGlobalJniEnv(jniEnv) == FALSE) {
+			SLOGE("Failed to set global jni env");
```
Comment (ja2-derek) [inline src/sgp/SGP.cc:354]: What would happen if we continue we `jniEnv` being null? I mean should we be throwing an exception if it is not going to work?

Comment (selaux) [reply src/sgp/SGP.cc]: Good point. Fixed now (and also when failing to find home directory).

**Comment by selaux** [comment]
Any remaining comments about this or can I merge?

**Comment by lynxlynxlynx** [comment]
LGTM!

**Pull request merged**